### PR TITLE
Adding date filtering feature and other improvements

### PR DESCRIPTION
**Pull request by Vicbi**
# Adding date filtering feature and other improvements

## :gear: Release Notes 
*Incorporation of the date filtering feature for the visualisation tool.

## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: SpeziDataPipeline.ipynb [hunk 1]
```diff
@@ -36,11 +37,13 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "serviceAccountKey_file = 'path/to/serviceAccountKey.json'\n",
+    "serviceAccountKey_file = 'serviceAccountKey_files/cs342-2024-example-application-firebase-adminsdk-z4ske-800739b1af.json'\n",
```
Comment (PSchmiedmayer) [inline SpeziDataPipeline.ipynb:40]: Might make sense to keep this generic without a specific example path in here for the open-source version.

**Review by PSchmiedmayer — APPROVED** [review]
Thank you for the changes; I just reviewed the code and it is great that you added some more context and surfaced more of the core functionality.

I only had a minor comment that the file name should stay generic in the open-source version of the notebook and we should not contain any project-specific code in there.

File: SpeziDataPipeline.ipynb (hunk 1 above)
Comment (Vicbi) [reply SpeziDataPipeline.ipynb]: Agree! Updated in the revised code.

File: README.md [hunk 2]
```diff
@@ -1 +1,125 @@
...
+daily_df = calculate_daily_data(filtered_df)
+```
+
+### 5. Create visual representations to explore the data.
+
+```python
+vizualize_data(daily_df)
+
+```
+![visualize_data.png](attachment:b0a17b62-3676-4abb-bcb5-643f623c96ce.png)
```
Comment (PSchmiedmayer) [inline README.md:111]: I think this file would need to be put in the repository to put it in the README. Attachments only work on issues and PR comments/descriptions: https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/attaching-files.

File: README.md [hunk 3]
```diff
@@ -1 +1,125 @@
...
+    # Initialize the Firebase app with the credentials
+    firebase_admin.initialize_app(cred)
+
+# Create a Firestore client instance to interact with the database
+db = firestore.client()
+```
+Notes:
+
+- Replace 'path/to/your/serviceAccountKey.json' with the actual path to the .json file you downloaded earlier.
+- The 'projectId' is your Firebase project ID, which you can find in your Firebase project settings.
```
Comment (PSchmiedmayer) [inline README.md:82]: You can use alerts to highlight elements in your README or any MD file, might be a good use here: https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts

File: README.md [hunk 4]
```diff
@@ -1 +1,125 @@
...
+
+**[matplotlib](https://matplotlib.org)** 
+
+**[firebase_admin](https://firebase.google.com/docs/admin/setup)** 
+
+You can install all required external packages using pip, which is the package installer for Python, with the following command:
+
+`bash
+pip install pandas numpy matplotlib firebase-admin
+`
```
Comment (PSchmiedmayer) [inline README.md:43]: I think this formatting is off:
````suggestion
```bash
$ pip install pandas numpy matplotlib firebase-admin
```
````

File: README.md [hunk 5]
```diff
@@ -1 +1,125 @@
...
+
+Contributions to this project are welcome. Please make sure to read the [contribution guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md) and the [contributor covenant code of conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) first.
+
+
+## License
+
+This project is licensed under the MIT License. See [Licenses](https://github.com/StanfordSpezi/SpeziAccessGuard/tree/main/LICENSES) for more information.
+
+![Spezi Footer](https://raw.githubusercontent.com/StanfordSpezi/.github/main/assets/FooterLight.png#gh-light-mode-only)
+<!-- ![Spezi Footer](https://raw.githubusercontent.com/StanfordSpezi/.github/main/assets/FooterDark.png#gh-dark-mode-only) -->
```
Comment (PSchmiedmayer) [inline README.md:125]: I think this line should be uncommented.

**Review by PSchmiedmayer — CHANGES_REQUESTED** [review]
I had some comments for the README to improve the formatting and some issues in the formatting how this will be displayed in the repo 👍

File: README.md (hunk 2 above)
Comment (Vicbi) [reply README.md:111]: The file is added.

File: README.md (hunk 4 above)
Comment (Vicbi) [reply README.md:43]: Corrected.

File: README.md (hunk 5 above)
Comment (Vicbi) [reply README.md]: 👍

**Pull request merged**